### PR TITLE
#159053961 Articles feature implementation 

### PR DESCRIPTION
**Pull request by johnmusiu**
#### What does this PR do?
Implements create, read, update and delete article(s).

#### Description of tasks to be completed
List of implemented article features:
1. A user can create an article.
2. A user can edit and/or delete an article they created.
3. Any user (authenticated or not)  can view any article by `slug`.
4. Any user (authenticated or not) can view a list of all articles.

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans/tree/ft-articles-crud-159053961
```
```sh
$ cd ah-titans
```
> Create a virtual environment and activate it.

```sh
$ virtualenv -p python3 venv
```
```sh
$ source venv/bin/activate
```

> Add **.env** file and copy **example.env** content.

``` cp example.env .env```

> Add your local database configuration settings to your **.env** file

```sh
$ source .env
```

```sh
$ pip install -r requirements.txt
```

```sh
$ python manage.py makemigrations
```

```sh
$ python manage.py migrate
```

```sh
$ python manage.py runserver
```

Now you can test the various endpoint on the postman.

#### Any background context you want to provide?
None.

#### What are the relevant Pivotal Tracker Stories?
[#159053961](https://www.pivotaltracker.com/story/show/159053961)

image

> View all articles
image

> View 1 article (belonging to the logged in user)
image

> View 1 article (not belonging to logged in user). It is similar to what an unlogged in user may see
image

### CODE REVIEW

**Comment by migot01** [comment]
Nice UI and  test coverage

**Comment by chepmanu** [comment]
Nice implementation on the article features  and good UI too, the color choice is simple and does not entail a lot of different colors.

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        # Validate title is not a series of symbols or non-alphanumeric characters
+        if re.match(r"[!@#$%^&*~\{\}()][!@#$%^&*~\{\}()]{2,}", title):
+            raise serializers.ValidationError(
+                """A title must not contain two symbols/foreign characters following each other"""
+            )
+        # Validate the description is not a series of symbols or
+        # non-alphanumeric characters
+        if re.match(r"[!@#$%^&*~\{\}()][!@#$%^&*~\{\}()]{2,}", description):
+            raise serializers.ValidationError(
+                """A description must not contain two symbols/foreign characters following each other"""
```
Comment (ridgekimani) [inline authors/apps/articles/serializers.py]: Indent this

Comment (johnmusiu) [reply authors/apps/articles/serializers.py]: I am not sure what you suggest I do about this line

**Comment by janetnim** [comment]
Good work on testing the feature, You have looked into all the edge cases as should be.

**Review by ridgekimani — APPROVED** [review]

**Pull request merged**